sync.sh

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: sync [-p] [-s] [--branch <branch>] [--merge] [--rebase]"
+    echo "Repository syncing script."
+    echo
+    echo "  -p         Restore all NuGet packages for the repository"
+    echo "  -s         Update source by fast-forwarding upstream -> origin -> local"
```

Comment:
I think this should just do a fetch like we do in sync.cmd the default actually mutating any of my branch state. It definitely makes too many assumptions about the names of remotes and structure of the branch in its current state.